.tekton/wait-for-image-task.yaml

```diff
@@ -10,6 +10,10 @@ spec:
     description: Image reference.
     type: string
   results:
+  - name: GIT_REF
+    description: Git commit for the image's source code if provided as `vcs-ref` image label.
+  - name: GIT_REPO
+    description: Git repository URL for the image's source code if provided as `source-location` image label.
```

Comment:
I keep thinking about this and I keep not liking it. The thing is: unlike `vcs-ref` that's provided by Konflux automatically, `source-location` is our stuff and it looks not tied to anything in the Dockerfiles. I imagine in a future some engineer may ask themselves "what is this doing here?" and kill the label. Either we need to guard the labels in comments or what may be much simpler is hardcode git repos in the pipeline. We already have params for quay repos, adding git repos wouldn't make it much worse.

WDYT?

Reply:
I want to avoid hardcoding in the pipeline as much as possible. 

Ideal scenario: We can guard in the script against git revision and repository (LABEL) being empty strings and fail the pipeline. 
If someone would remove the LABEL, we'd notice in the `operator-bundle` failure. 

That would certainly work for the monorepo components and PRs that update the SCANNER/COLLECTOR version - if all engineers would follow up on Konflux build pipeline faillures.

Do you have an explanation why `vcs-url` is not a label required by EC?
https://github.com/release-engineering/rhtap-ec-policy/blob/main/data/rule_data.yml#L76C1-L83C46

Reply:
> Do you have an explanation why `vcs-url` is not a label required by EC?

No idea. Some other labels rules seem quite random to me.

>  If someone would remove the LABEL, we'd notice in the operator-bundle failure.

Yes, that'll be helpful but not ideal. Folks removing the label in Collector or Scanner repos won't see the effect of this.
I guess, let's just do it (if you like it more than hardcoding repos). I'd also ask you to put comments in each konflux.*Dockerfile next to the `source-location` label saying it must not be removed or renamed because it's used for snapshot creation.

Reply:
Validating the component object in the Python script: https://github.com/stackrox/stackrox/pull/13577/commits/d9202fa536edc937e4795b8960af8627617b8559.

Created https://issues.redhat.com/browse/ROX-27485 to add the comment to the Collector and Scanner v2 Dockerfiles. We don't use the source-location in the components built from this repository, so we could remove the label without breaking the snapshot creation.

Reply:
Continuing here: https://github.com/stackrox/stackrox/pull/13577#discussion_r1894304576